.github/workflows/python-publish.yml

```diff
@@ -52,6 +52,19 @@ jobs:
         with:
           pythonVersion: "3.11"
           dependencyType: "docs"
+      - name: Append release notes to RELEASE_NOTES.md # automatically update release notes on new release
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git checkout main
```

Comment:
We're checking out code in the step above, do we need this step again?

Reply:
Yes here we are not repeating the checkout step, So in the checkout step above we are using GitHub actions to checkout our source code into a runner for our workflow, while here we are using git checkout to make sure that the changes in `RELEASE_NOTES.md` are correctly pushed to the main branch.
When a new release is created and the workflow is triggered `git checkout main` makes sure that the branch being worked on is `main`, it makes sure there is no ambiguity about the branch.
Hope that answer's it. I added a comment earlier, I think I deleted that by mistake :(